### PR TITLE
Disable Rails console IRB's autocompletion in production by default.

### PR DESCRIPTION
**Pull request by st0012**
### Motivation / Background

1. Autocompletion increases data transmission significantly. This could cause noticeable slowdown when connecting to remote servers, which is usually the case in production.
2. The autocompletion feature still has many issues, as listed in https://github.com/ruby/irb/issues/445. They may be just annoying when happened locally, but in production they could cause real issues (e.g. typos caused by slow backspacing).

Due to these reasons, I think it's safer to disable this feature in production.

### Detail

Because `IRB.start` doesn't take configuration arguments and rebuilds the `IRB.conf` object, the only way we can turn off autocompletion is through the `IRB_USE_AUTOCOMPLETE` env var.

The env var was added in https://github.com/ruby/irb/pull/469 and will be available in IRB 1.6+ and Ruby 3.2+. The name wasn't used before so it won't cause issues with older IRB versions.

### Additional information

Users can still override this by setting `IRB_USE_AUTOCOMPLETE=true`.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

**Comment by eileencodes** [comment]
Thank you @st0012! 

Backported to 7-0-stable in cc125a0f9a894baa4e33f92995086897bf09da9d and 7163a132d2ee552c7e601665448d7f52f111c057.

**Comment by jarthod** [comment]
Thanks @st0012, I complained a lot about this completion when connecting to production servers (and locally to a smaller extent) and wondered if I was the only one, apparently not :)

**Comment by krtschmr** [comment]
great PR. thanks!

File: railties/lib/rails/commands/console/console_command.rb [hunk 1]
```diff
@@ -38,6 +38,10 @@ def initialize(app, options = {})
 
       if @console == IRB
         IRB::WorkSpace.prepend(BacktraceCleaner)
+
+        if Rails.env == "production"
```
Comment (simi) [inline railties/lib/rails/commands/console/console_command.rb:42]: Any reason to not use `Rails.env.production?` or even reverse the logic and set this only in development so all other environments (for example I have seen `staging` environment used often) will not be affected by completion?

Comment (krtschmr) [reply railties/lib/rails/commands/console/console_command.rb:42]: i would agree with the idea of reversing it and only enable it by default on development. 
in a perfect world, i would disable it for all environments and let the dev decide when to enable it, explicitly. either via `development.rb` or ENV var

**Pull request merged**